zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java

```diff
@@ -405,37 +409,36 @@ public HPerson getLastTranslator(HProject project) {
-                                "from HProject p " +
-                                "where :maintainer in elements(p.maintainers) " +
-                                "and p.status <> " +
-                                "'" + EntityStatus.OBSOLETE.getInitial() + "' " +
-                                "and (p.name like :filter " +
-                                    "or p.slug like :filter) " +
-                                "order by p.name")
-                        .setParameter("maintainer", maintainer)
-                        .setParameter("filter", "%" + sqlFilter + "%")
-                        .setFirstResult(firstResult)
-                        .setMaxResults(maxResults);
+        Query q = getSession().createQuery(
+                "select m.project from HProjectMember m " +
+                        "where m.person = :maintainer " +
+                        "and m.role = :role " +
+                        "and m.project.status <> :obsolete " +
+                        "and (m.project.name like :filter " +
+                        "or m.project.slug like :filter) " +
+                        "order by m.project.name")
+                .setParameter("maintainer", maintainer)
+                .setParameter("role", ProjectRole.Maintainer)
+                .setParameter("obsolete", EntityStatus.OBSOLETE)
+                .setParameter("filter", "%" + sqlFilter + "%")
+                .setFirstResult(firstResult)
+                .setMaxResults(maxResults);
         return q.list();
     }
 
     public int getMaintainedProjectCount(HPerson maintainer, String filter) {
         final String sqlFilter = filter == null ? "" : filter;
-        Query q =
-                getSession()
-                        .createQuery(
-                                "select count(p) from HProject p " +
-                                "where :maintainer in elements(p.maintainers) " +
-                                "and p.status <> " +
-                                "'" + EntityStatus.OBSOLETE.getInitial() + "' " +
-                                "and (p.name like :filter " +
-                                    "or p.slug like :filter) "
-                        )
-                        .setParameter("maintainer", maintainer)
-                        .setParameter("filter", "%" + sqlFilter + "%");
+        Query q = getSession().createQuery(
+                "select count(m) from HProjectMember m " +
+                        "where m.person = :maintainer " +
+                        "and m.role = :role " +
+                        "and m.project.status <> :obsolete " +
+                        "and (m.project.name like :filter " +
+                        "or m.project.slug like :filter)")
+                .setParameter("maintainer", maintainer)
+                .setParameter("role", ProjectRole.Maintainer)
+                .setParameter("obsolete", EntityStatus.OBSOLETE)
+                .setParameter("filter", "%" + sqlFilter + "%");
```

Comment:
Just the filters that were already there - the only real change here is to make the query work with the new database structure for maintainers, so I tried to keep the behaviour the same.